app/controllers/application_controller.rb

```diff
@@ -607,7 +607,15 @@ def determine_custom_metadata_keys
       metadata_type = CustomMetadataType.find(attribute_params[:custom_metadata_type_id])
       if metadata_type
         keys = [:custom_metadata_type_id]
-        keys = keys + [{data:[metadata_type.custom_metadata_attributes.collect(&:title)]}]
+        cma= []
+        metadata_type.custom_metadata_attributes.each do |attr|
+          if attr.sample_attribute_type.base_type == Seek::Samples::BaseType::LIST
+            cma << {attr.title=>[]}
+          else
```

Comment:
Is this needed? How does SeekSampleMulti work?

Reply:
the params of `custom_metadata_attributes`
`"custom_metadata_attributes"=>#<ActionController::Parameters {"custom_metadata_type_id"=>"7", "data"=>{"name"=>"a", "resource_type_general"=>"Audiovisual", "resource_type_general_list"=>["Book"], "resource_study_country_list"=>["Algeria", "Andorra"]}} permitted: false>,`

- before change of my code

```
determine_custom_metadata_keys =
`=> [:custom_metadata_type_id, {:data=>[["name", "resource_type_general", "resource_type_general_list", "resource_study_country_list"]]}]`
```

- after change 

`determine_custom_metadata_keys => [:custom_metadata_type_id, {:data=>[["name", "resource_type_general", {"resource_type_general_list"=>[]}, {"resource_study_country_list"=>[]}]]}]`



And for samples 

```
sample_type_param_keys = sample_type ? sample_type.sample_attributes.map(&:title).collect(&:to_sym) : []
=> [:name, :"linked sample"] 
```
linked sample is just the id of the linked sample